From 7bd0146ec173560efe80a9c64646766a13982ae5 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 2 Feb 2021 23:58:57 +0000
Subject: [PATCH 2/5] gnu: Add cl-slug

* gnu/packages/lisp-xyz.scm
  (sbcl-cl-slug, cl-slug, ecl-cl-slug): New variables
---
 gnu/packages/lisp-xyz.scm | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index 7957b50490..204c0ca0b6 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -6650,6 +6650,43 @@ of C+GObject libraries without the need of writing dedicated bindings.")
 (define-public ecl-cl-gobject-introspection
   (sbcl-package->ecl-package sbcl-cl-gobject-introspection))
 
+(define-public sbcl-cl-slug
+  (let ((commit "ffb229d10f0d3f7f54e706791725225e200bf749")
+        (revision "1"))
+    (package
+      (name "sbcl-cl-slug")
+      (version (git-version "0.4.1" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/EuAndreh/cl-slug")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "1asdq6xllmsvfw5fky9wblqcx9isac9jrrlkfl7vyxcq1wxrnflx"))))
+      (build-system asdf-build-system/sbcl)
+      (arguments
+       `(#:asd-files '("cl-slug-test.asd" "cl-slug.asd")
+         #:asd-systems '("cl-slug-test" "cl-slug")))
+      (native-inputs
+       `(("prove" ,sbcl-prove)))
+      (inputs
+       `(("ppcre" ,sbcl-cl-ppcre)))
+      (home-page "https://github.com/EuAndreh/cl-slug")
+      (synopsis "Multy languages slag formater")
+      (description
+       "Small library to make slugs, mainly for URIs, from english and beyond.
+
+This package provides 1 system: @code{CL-SLUG}")
+      (license license:expat))))
+
+(define-public ecl-cl-slug
+  (sbcl-package->ecl-package sbcl-cl-slug))
+
+(define-public cl-slug
+  (sbcl-package->cl-source-package sbcl-cl-slug))
+
 (define-public sbcl-string-case
   (let ((commit "718c761e33749e297cd2809c7ba3ade1985c49f7")
         (revision "0"))
-- 
2.30.0

